tracer/src/Datadog.Trace/ReadOnlySpanContext.cs

```diff
@@ -7,17 +7,31 @@ namespace Datadog.Trace
 {
     internal class ReadOnlySpanContext : ISpanContext
     {
-        public ReadOnlySpanContext(ulong traceId, ulong spanId, string serviceName)
+        public ReadOnlySpanContext(TraceId traceId, ulong spanId, string serviceName)
         {
             TraceId = traceId;
             SpanId = spanId;
             ServiceName = serviceName;
         }
 
-        public ulong TraceId { get; }
+        /// <summary>
+        /// Gets the lower 64 bits of the 128-bit trace identifier.
+        /// </summary>
+        ulong ISpanContext.TraceId => TraceId.Lower;
 
+        /// <summary>
+        /// Gets the 128-bit trace identifier.
+        /// </summary>
+        public TraceId TraceId { get; }
```

Comment:
To match the `SpanContext`, could this property be renamed to `TraceId128` and then the `TraceId` property could remain the same?

Reply:
Ahh yes. I had a little back and forth before settings on `TraceId128`. I forgot to update this one, thanks!

Reply:
Done.